jcstress-core/src/main/java/org/openjdk/jcstress/TimeBudget.java

```diff
@@ -131,10 +134,46 @@ public void printOn(PrintStream out) {
         } else {
             out.println("    Initial completion estimate: " + ReportUtils.msToDate(timeLeftMs(), true));
             out.println("    Initial test time: " + targetTestTimeMs() + " ms");
+            printOvertimeWarning();
         }
         out.println();
     }
 
+    private long countEta(int msPerTest) {
+        long expectedTotalTime = (long) expectedTests * msPerTest;
+        long expectedPerTest = expectedTotalTime / getConcurentTestsFactor();
+        return expectedPerTest;
+    }
+
+    private boolean printOvertimeWarning() {
+        long expectedPerTest = countEta(DEFAULT_PER_TEST_MS);
+        boolean print=false;
+        if (expectedPerTest > budget.milliseconds() * 2l) {
+            System.out.println(" + +++ FATAL - your tests will never finish as expected. They will run much longer ");
```

Comment:
I don't think you should assume `System.out`. All these prints should be on a given `PrintStream out`, see `printOn()` method.

The formatting of the message does not seem to adhere to other messages. You should see how messages are printed elsewhere in the code and try to use same format? E.g.

```
out.println("FATAL: No matching tests.")
```

Also, about this particular message, if this is fatal jcstress should not run. I don't think this is really happening here. The `print=true` does not cause jcstress not to run.

Reply:
Damn, sorry for this.  This was changed globa-wide recently, and I had kept it correctly in all prs. Ibviously I had become less vigilant at the end, sorry. Fixed here and also in one  more PR which was affected.

Reply:
> ```
> out.println("FATAL: No matching tests.")
> ```
> 
> Also, about this particular message, if this is fatal jcstress should not run. I don't think this is really happening here. The `print=true` does not cause jcstress not to run.

forget to reply this. Yes. it will die correctly later.